game-core/src/main/java/games/strategy/triplea/ui/MapPanel.java

```diff
@@ -568,7 +570,9 @@ public void paint(final Graphics g) {
       g2d.drawImage(mouseShadowImage, t, this);
     }
     if (routeDescription != null) {
-      routeDrawer.drawRoute(g2d, routeDescription, movementLeftForCurrentUnits);
+      routeDrawer.drawRoute(g2d, routeDescription, movementLeftForCurrentUnits, movementFuelCost,
```

Comment:
It would be nice if drawRoute would continue to only accept 3 arguments, the third one being a wrapper object to wrap the last 3 objects.
This would keep the related stuff together and the parameter declarations relatively clean.

Reply:
I think as a whole there could be some refactoring of parameters across a lot of the MapRouteDrawer and MapPanel. Both in terms of wrapper objects and types used. I decided not to dive into that for this PR. I think looking at even existing objects like RouteDescription which might be expanded to include these. But for now, I think that it would be better in a separate PR.